File: src/main/java/org/micoli/php/symfony/list/CommandElementDTO.java
package org.micoli.php.symfony.list;

import com.intellij.psi.PsiElement;
import java.util.List;
import org.jetbrains.annotations.NotNull;

public record CommandElementDTO(
        @NotNull String command, @NotNull String description, @NotNull String className, @NotNull PsiElement element)
        implements SearchableRecord {

    @Override
    public List<@NotNull String> getSearchString() {
        return List.of(command, description, className);
    }
}

File: src/main/kotlin/org/micoli/php/symfony/list/CommandElementDTO.kt
package org.micoli.php.symfony.list

import com.intellij.psi.PsiElement
import kotlinx.collections.immutable.ImmutableList
import kotlinx.collections.immutable.persistentListOf

@JvmRecord
data class CommandElementDTO(
    val command: String,
    val description: String,
    val className: String,
    val element: PsiElement
) : SearchableRecord {
    override fun getSearchString(): ImmutableList<String> {
        return persistentListOf(command, description, className)
    }
}
